test/SolidDaoManagement.t.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "forge-std/Test.sol";
+import "../contracts/SolidDaoManagement.sol";
+
+contract SolidDaoManagementTest is Test {
+
+    event GovernorPushed(address indexed from, address indexed to, bool _effectiveImmediately);
+    event GuardianPushed(address indexed from, address indexed to, bool _effectiveImmediately);    
+    event PolicyPushed(address indexed from, address indexed to, bool _effectiveImmediately);    
+    event VaultPushed(address indexed from, address indexed to, bool _effectiveImmediately);    
+
+    event GovernorPulled(address indexed from, address indexed to);
+    event GuardianPulled(address indexed from, address indexed to);
+    event PolicyPulled(address indexed from, address indexed to);
+    event VaultPulled(address indexed from, address indexed to);
+
+    SolidDaoManagement private solidDaoManagement;
+
+    address internal governorAddress = vm.addr(1);
+    address internal guardianAddress = vm.addr(2);
+    address internal policyAddress = vm.addr(3);
+    address internal vaultAddress = vm.addr(4);
+
+    address internal newGovernorAddress = vm.addr(5);
+    address internal newGuardianAddress = vm.addr(6);
+    address internal newPolicyAddress = vm.addr(7);
+    address internal newVaultAddress = vm.addr(8);
+
+    address internal otherAddress = vm.addr(9);
+    
+    function setUp() public {
+        solidDaoManagement = new SolidDaoManagement(
+            governorAddress,
+            guardianAddress,
+            policyAddress,
+            vaultAddress
+        );
+    }
+
+    function testGovernorAddress() public {
+        assertEq(solidDaoManagement.governor(), governorAddress);
+    }
+
+    function testGuardianAddress() public {
+        assertEq(solidDaoManagement.guardian(), guardianAddress);
+    }
+
+    function testPolicyAddress() public {
+        assertEq(solidDaoManagement.policy(), policyAddress);
+    }
+
+    function testVaultAddress() public {
+        assertEq(solidDaoManagement.vault(), vaultAddress);
+    }
+
+    function testPushGovernor() public {
+        vm.prank(governorAddress);
+        vm.expectEmit(true, true, true, true);
+        emit GovernorPushed(newGovernorAddress, newGovernorAddress, true);
+        solidDaoManagement.pushGovernor(newGovernorAddress, true);
+        assertEq(solidDaoManagement.governor(), newGovernorAddress);
+        assertEq(solidDaoManagement.newGovernor(), newGovernorAddress);
+    }
+
+    function testUnautorizedPushGovernor() public {
+        vm.expectRevert(bytes("UNAUTHORIZED"));
+        vm.prank(otherAddress);
+        solidDaoManagement.pushGovernor(newGovernorAddress, true);
+        assertEq(solidDaoManagement.governor(), governorAddress);
+    }
+
+    function testPullGovernor() public {
+        vm.prank(governorAddress);
+        vm.expectEmit(true, true, true, true);
+        emit GovernorPushed(governorAddress, newGovernorAddress, false);
```

Comment:
Would not be better to emit this event this after the action method tested (in this case pushGovernor) be performed ?